File: 2st/Riyeon/겹치는 선분의 길이.py
def solution(lines):
    s1 = set()
    s2 = set()
    s3 = set()
    for i in range(lines[0][0], lines[0][1]):
        s1.add(i)
    for i in range(lines[1][0], lines[1][1]):
        s2.add(i)
    for i in range(lines[2][0], lines[2][1]):
        s3.add(i)
    return len((s1 & s2) | (s1 & s3) | (s2 & s3))

File: 4st/Eunseong/겹치는 선분의 길이.java
class Solution {
    public int solution(int[][] lines) {
        int numline[] = new int[201];
        for (int i = 0; i < lines.length; i++) {
            for (int j = lines[i][0] + 100; j < lines[i][1] + 100; j++) {
                numline[j]++;
            }
        }
        int truecnt = 0;
        for (int i : numline) {
            if (i >= 2) truecnt++;
        }
        return truecnt;
    }
}